01_Exercises/LAB_03.py

```diff
@@ -0,0 +1,171 @@
+    random_list = do_list_integers(random_nr)
+    prime_numbers = []
+
+    for num in random_list:
+        if num > 1:
+            for i in range(2, num):
+                if (num % i) == 0:
+                    break
+            else:
+                prime_numbers.append(num)
+    return prime_numbers
+
+
+assert find_prime_numbers("234910") == [2, 3]
+assert find_prime_numbers("-157384") == [5, 7, 3]
+assert find_prime_numbers("-10245") == [2, 5]
+assert find_prime_numbers("984132") == [3, 2]
+assert find_prime_numbers("486") == []
+
+
+def mountain_sequence_pattern(random_nr):
+    """
+    This function can take 5 numbers;
+    :param random_nr: must be a random number or a string number.
+    :return: FALSE OR TRUE - and - descriptive string.
+    """
+    random_list = do_list_integers(random_nr)
+    sequence_one = 0
+    sequence_two = 0
+
+    if len(random_list) > 5 or len(random_list) < 5:
+        return False
+
+    for num in range(len(random_list) - 3):
+        if random_list[num] + 1 == random_list[num + 1]:
+            sequence_one += 1
+        else:
+            return False
+
+    for other_num in range(2, len(random_list) - 1):
+        if random_list[other_num] - 1 == random_list[other_num + 1]:
+            sequence_two += 1
+        else:
+            return False
+
+    if sequence_one == sequence_two:
+        return True
+
+
+assert mountain_sequence_pattern("234592") == False
+assert mountain_sequence_pattern("-1010-1") == True
+assert mountain_sequence_pattern("12321") == True
+assert mountain_sequence_pattern("48238") == False
+assert mountain_sequence_pattern("3") == False
+
+
+def main_menu():
+    """
+    :return: A list of options/strings.
+    """
+    menu = ["Main Menu", "Option 1: To generate a list of integers input 1", "Option 2: To find prime numbers in a list input 2", "Option 3: To detect a mountain sequence number input 3"]
+
+    for options in menu:
+        print(options)
+
+    return menu
+
+
+def return_to_menu(menu_or_exit):
+    """
+    This function takes the user input to exit or to return to the main menu.
+    """
+    if menu_or_exit == "main menu":
+        return main()
+    else:
+        print("You've exit the application")
+        pass
+
+
+def main():
+    """
+    This is the main function of the app which provides instructions and it takes a random number input.
+    Examples:
+        1213402
+        -2-13-49 etc
+    """
+    main_menu()
+    selected_option = input("Choose your option here: ")
+
+    if selected_option == "1":
+        os.system("cls")
+        print("You've choose option 1: Generate a list of integers")
+        user_input = input("Write some numbers and the app will convert it into a list of integers: ")
+        user_result = do_list_integers(user_input)
+        print("Output:")
+
+        for num in user_result:
+            print(num, end=" ")
+        return user_result, return_to_menu(input("\nWrite main menu to return to it or exit: "))
```

Comment:
Ca idee , returnezi 2 valori degeaba si dai `input("\nWrite main menu to return to it or exit: ")` ca acelasi parametru peste tot.
Nu trebuie sa returnezi nicio valoare ca idee pentru ca deja ai folosit valoarea la ce trebuia sa o folosesti, ai afisat-o pe ecran. Poti scoate returnul si sa muti inputul inauntrul functiei `return_to_menu` pe care o apelezi o singura data la finalul lantului de `if, elif elif...`

Reply:
True. Am modificat chestia asta si am mutat-o in functie.